### PR TITLE
Prevent null exception with columns using "object.inner.value" keys

### PR DESCRIPTION
**Pull request by RichTeaTime**
### Context
When reading a column description of the form "object.inner.value", with data of form e.g. 
[ { inner:{ value:2 } }, { inner: null } ]
getProperty and DataMap.get would return an exception 'Cannot read property 'value' of null' on the second row.

Expanded the existing "=== undefined" (void 0) check to also include "=== null", as this is similarly defensive and should not be trigger an exception.

### How has this been tested?
Tested with null and undefined values, and undefined members in column keys for multiple datasets.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Checklist:
- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/dataMap.js [hunk 1]
```diff
@@ -594,7 +594,7 @@ class DataMap {
       for (let i = 0, ilen = sliced.length; i < ilen; i++) {
         out = out[sliced[i]];
 
-        if (typeof out === 'undefined') {
+        if (out === undefined || out === null) {
```
Comment (wojciechczerniak) [inline src/dataMap.js]: We prefer `void 0` over `undefined` so without `typeof` it should look like:

```suggestion
        if (out === void 0 || out === null) {
```

Thanks for the PR!

**Comment by AMBudnik** [comment]
Thank you for sharing @RichTeaTime 

I do not know if it's the code (it would be great if you could add tests) or the Travis but restarting tests doesn't help.

File: src/dataMap.js (hunk 1 above)
Comment (pnowak) [inline src/dataMap.js]:
```suggestion
        if (out === null) {
          /* eslint-disable-next-line no-continue */
          continue;
        }

        if (out === void 0) {
          return null;
        }
```

File: src/dataMap.js [hunk 2]
```diff
@@ -594,7 +594,7 @@ class DataMap {
       for (let i = 0, ilen = sliced.length; i < ilen; i++) {
         out = out[sliced[i]];
```
Comment (pnowak) [inline src/dataMap.js]: So, maybe the better thing is the first check `out` and if it's not `falsy` use it. With this improvement, we don't need the second check, so inner `if` is useless.
```suggestion
        out = out && out[sliced[i]];
```

**Comment by AMBudnik** [comment]
This pull request has to be closed as it does not meet all the Contributing Guidelines listed at https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md so it has to be closed and hasn't been updated for a while.

**Pull request closed without merging**